assets/src/edit-story/app/prepublish/getPrepublishErrors.js

```diff
@@ -0,0 +1,126 @@
+ * limitations under the License.
+ */
+/**
+ * Internal dependencies
+ */
+import error from './error';
+import warning from './warning';
+import guidance from './guidance';
+
+/**
+ *
+ * @typedef {import ('./types').Guidance} Guidance
+ * @typedef {import('../../types').Story} Story
+ */
+
+/**
+ *
+ * @param {Story} initialStoryState The full story object being checked for guidance messages.
+ * @return {Guidance[]} The array of checklist items to be rectified.
+ */
+
+function getPrepublishErrors(story) {
+  if (!story) {
+    return [];
+  }
+  const checklistResult = [error, warning, guidance]
+    .map((byType) => {
+      const {
+        // arrays of functions
+        story: storyChecklist = [],
+        page: pageChecklist = [],
+        ...elementChecklistsByType
+      } = byType;
+
+      let storyGuidance = [],
+        pageGuidance = [],
+        elementGuidance = [];
+
+      // get guidance messages for the top-level story object
+      storyChecklist.forEach((getStoryGuidance) => {
+        try {
+          const guidanceMessage = getStoryGuidance(story);
+          if (guidanceMessage !== undefined) {
+            storyGuidance.push(guidanceMessage);
+          }
+        } catch (e) {
+          // ignore errors
+        }
+      });
+
+      // for each page, run checklist on the page object as well as each element
+      ({ pageGuidance, elementGuidance } = story.pages.reduce(
+        (prev, currentPage, currentIndex) => {
+          const page = currentIndex + 1;
+          const { id: pageId, elements } = currentPage;
+
+          // get guidance for the page object
+          const currentPageGuidance = [];
+          pageChecklist.forEach((getPageGuidance) => {
+            try {
+              const guidanceMessage = getPageGuidance(currentPage);
+              if (guidanceMessage !== undefined) {
+                currentPageGuidance.push({ ...guidanceMessage, page });
+              }
+            } catch (e) {
+              // ignore errors
+            }
+          });
+
+          // get guidance for all the elements on the page
+          let currentPageElementGuidance = [];
+          elements.forEach((element) => {
+            const elementsChecklist =
+              elementChecklistsByType[element.type] || [];
+
+            elementsChecklist.forEach((getElementGuidance) => {
+              try {
+                const guidanceMessage = getElementGuidance(element);
+                if (guidanceMessage !== undefined) {
+                  currentPageElementGuidance.push({
+                    ...guidanceMessage,
+                    // provide the page the element is on
+                    pageId,
+                    page,
+                  });
+                }
+              } catch (e) {
+                // ignore errors
+              }
+            });
+          });
+
+          return {
+            pageGuidance: [...prev.pageGuidance, ...currentPageGuidance],
```

Comment:
Is there ever a case these could not be iterable?

Reply:
No, I don't think so. `prev.pageGuidance` will be initialized as an array due to the provided default value in the second `reduce` param. `currentPageGuidance` is also initialized as an array and only modified with `push`ing. :+1: